crypto/rand/rand_crng_test.c

```diff
@@ -64,27 +65,29 @@ static const OPENSSL_CTX_METHOD rand_crng_ossl_ctx_method = {
 };
 
 int rand_crngt_get_entropy_cb(OPENSSL_CTX *ctx,
+                              RAND_POOL *pool,
```

Comment:
What is the rationale behind the changes in `rand_crng_test.c`?

Reply:
Is it a fixup for 4e297b7441a0 which you made along the way?

Reply:
Effectively yes. I encountered a crash related to this code. The change is related to the `crngt_get_entropy` callback. That callback gets called from `rand_crng_ossl_ctx_new`. That function is called when we are initialising an OPENSSL_CTX with CRNG data for the first time. Unfortunately, in the original implementation, the default callback was itself trying to get CRNG data from the OPENSSL_CTX. But since we're still initialising it it went round in an an infinite recursive loop trying to initialise the OPENSSL_CTX until eventually it blew the stack. The solution to the problem was to change the callback to take the RAND_POOL to be used as a parameter instead of getting it from the OPENSSL_CTX.